File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution 
{
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) 
    {
        ListNode res = new ListNode(0);
        ListNode head = res;

        while(list1!=null && list2!=null)
        {
            if(list1.val < list2.val)//
            {
                head.next = new ListNode(list1.val);// 0 1 1 2
                list1 = list1.next;
            }
            else
            {
                 head.next = new ListNode(list2.val);// 0 1    0 1 1 2 3
                 list2 = list2.next;
            }
            head = head.next;

        }

         head.next = list1!=null ?  list1 :  list2;

    return res.next;
    }
}

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        ptr1: ListNode = list1
        ptr2: ListNode = list2

        head: ListNode = None;
        head = ListNode(-1);
        res: ListNode = head;
        curr: ListNode = None;

        while ptr1 is not None and ptr2 is not None:
            if ptr1.val < ptr2.val:
                curr = ListNode(ptr1.val)
                ptr1 = ptr1.next
            else:
                curr = ListNode(ptr2.val)
                ptr2 = ptr2.next
            
            res.next = curr
            res = res.next
        
        while ptr1 is not None:
            curr = ListNode(ptr1.val)
            ptr1 = ptr1.next
            res.next = curr
            res = res.next
        
        while ptr2 is not None:
            curr = ListNode(ptr2.val)
            ptr2 = ptr2.next
            res.next = curr
            res = res.next

        return head.next



        
        